merlin/systems/triton/models/oprunner_model.py

```diff
@@ -110,7 +117,10 @@ def execute(self, requests):
                         output_tensors.append(data)
                         continue
                     data = data.get() if hasattr(data, "get") else data
-                    tensor = pb_utils.Tensor(name, data)
+                    if not isinstance(data.values, np.ndarray):
+                        tensor = pb_utils.Tensor(name, cupy.asnumpy(data.values))
```

Comment:
What is the connection with DictArray/Column and this `cupy/numpy` conversion (and in the FIL op too)

Reply:
Previously we allowed any array type in `DictArray`, but since it's hard to build on top of the varying APIs of the different array/tensor libraries, we're heading in a direction where the only allowed types will be Numpy and CuPy arrays (which gives us a consistent API to build on.) That means for other types, like Triton tensors, we'll need to convert them across to Numpy or CuPy, hopefully via zero-copy transfers. Unfortunately, there's not currently a way to do that with Triton GPU tensors and CuPy arrays because their dlpack implementations are not compatible with each other. So we work around that here by forcing the data to be moved from GPU to CPU. It's a little gross, but it's the best way we've been able to figure out how to do it without using another library (e.g. PyTorch) as an intermediary.